src/no_std.rs

```diff
@@ -0,0 +1,92 @@
+//! Contains logic related to no_std functionaligy
+
+use core::fmt;
+
+/// Avoids the use of [`std::io::Cursor`] in `no_std` context
+pub(crate) struct Cursor<T> {
+    buffer: T,
+    position: usize,
+}
+impl<T> Cursor<T> {
+    pub(crate) fn new(buffer: T) -> Self {
+        Self {
+            buffer,
+            position: 0,
+        }
+    }
+    pub(crate) fn position(&self) -> usize {
+        self.position
+    }
+}
+impl<T: AsMut<[u8]>> fmt::Write for Cursor<T> {
+    fn write_str(&mut self, s: &str) -> fmt::Result {
+        let raw_s = s.as_bytes();
+        let pos = self.position;
+
+        let buf = &mut self.buffer.as_mut()[pos..pos + raw_s.len()];
+
+        if raw_s.len() > buf.len() {
+            return Err(fmt::Error);
+        }
+
+        buf.copy_from_slice(raw_s);
+        self.position += raw_s.len();
+
+        Ok(())
+    }
+}
+
+#[cfg(any(test, feature = "std"))]
+fn f64_significant_digits_std(value: f64) -> usize {
+    (value.abs().log10() + 1f64).floor() as usize
```

Comment:
I think this was incorrect previously. Please confirm

Reply:
Agree, the previous version can produce greater value, thanks for fixing.

Reply:
Could you help me make a test case in `src/tests/mod.rs::from_f64` where the conversion to fixed point results in incorrect precision? I'm not able to find the test case that covers this

Reply:
I think, tests in `tests::from_f64` are enough for now, I haven't found a counterexample in five minutes.

As an option, we can use `protest` and check something like the following property:
```rust
for f: f64 in (-1.0, 1.0):
    let round_trip = FixedPoint::try_from(f)?.try_into::<f64>()?;
    assert!((f - round_trip).abs() < 0.1.powi(FixedPoint::PRECISION.to_i32())
```

Reply:
Whatever you want, it's optional, LGTM now.

But if you want to get some experience in `proptest`, I don't want to stop you)